crates/shared/src/url.rs

```diff
@@ -12,3 +15,66 @@ pub fn join(url: &Url, mut path: &str) -> Url {
     }
     Url::parse(&format!("{url}/{path}")).unwrap()
 }
+
+/// Splits a URL right where the path begins into base and endpoint.
+/// https://my.solver.xyz/solve/1?param=1#fragment=some
+/// becomes
+/// (https://my.solver.xyz/, /solve/1?param=1#fragment=some)
+/// Path that were split like this can be joined to the original URL using
+/// [`join`].
+pub fn split_at_path(url: &Url) -> Result<(Url, String)> {
+    let base = format!(
```

Comment:
nit: If I'm not mistaken, this should be the same as `url.origin().unicode_serialization`

https://docs.rs/url/latest/url/struct.Url.html#method.origin

Reply:
Was looking for something like this but couldn't find it. 👍